File: python/src/Cola.py
class Nodo_Cola:
    def __init__(self, name=None, rut=None, age=0):
        self._name = name
        self._rut = rut
        self._age = age
        self._next = None

class Cola:
    def __init__(self):
        self.first = None
        self.last = None
        self.size = 0
    
    def is_empty(self):
        return self.first is None
    
    def get_size(self):
        return self.size

    def insert(self, nodo):
        if self.is_empty():
            self.first = nodo
            self.last = nodo
        else:
            self.last._next = nodo
            self.last = nodo
        
        nodo._next = None
        self.size += 1
    
    def unfold_queue(self):
        current = self.first
        
        if current is None:
            print("La cola está vacía.")
            return
        while current is not None:
            print(f"Nombre: {current._name}, RUT: {current._rut}, Edad: {current._age}")
            current = current._next
    
    def get_first(self):
        return self.first
    def get_last(self):
        return self.last

File: javascript/src/Cola.js
class QueueNode {
    constructor(name = null, rut = null, age = 0){
        this.name = name;
        this.rut = rut;
        this.age = age;
        this.next = null;
    }
}

export class Queue {
    constructor(){
        this.head = null;
        this.tail = null;
        this.size = 0;
    }
    isEmpty(){
        return this.size === 0;
    }
    enqueue(data){
        const node = { data, next: null};

        if(!this.head){
            this.head = this.tail = node;
        } else {
            this.tail.next = node;
            this.tail = node;
        }
        this.size++;
    }
    dequeue() {
        if(!this.head) return null;

        const data = this.head.data;
        this.head = this.head.next;
        this.size--;
        return data;
    }
    print(){
        if (this.isEmpty()){
            console.log("Queue is empty");
            return;
        }
        let current = this.head;
        while(current){
            console.log(`Name: ${current.name}, RUT: ${current.rut}, Age: ${current.age}`);
            current = current.next;
        }
    }
    toArray() {
        const result = [];
        let current = this.head;

        while (current) {
            result.push( current.data);
            current = current.next;
        }

        return result;
    }

    peek(){
        return this.head;
    }
}

//module.exports = {Queue, QueueNode};